build.sbt

```diff
@@ -61,9 +62,38 @@ def removeScalacOptionsInTest(scalaVersion: String) =
     case _ => Seq("-Wvalue-discard", "-Wunused:explicits", "-Wunused:params", "-Wunused:imports")
   }
 
+def macrosDepsByVersion(scalaVersion: String) =
+  CrossVersion.partialVersion(scalaVersion) match {
+    case Some((2, _)) =>
+      Seq(
+        "com.chuusai" %% "shapeless" % shapelessVersion,
+      )
+    case Some((3, _)) => 
+      Seq("org.typelevel" %% "shapeless3-deriving" % shapeless3Version)
+    case _ => Seq.empty  
+
+  }
+
+def maybeReflect(scalaVersion: String) =
+  CrossVersion.partialVersion(scalaVersion) match {
+    case Some((2, _)) =>
+      Seq(
+        "org.scala-lang" % "scala-reflect" % scalaVersion
+      )
+    case _ => Seq.empty  
+  }
```

Comment:
Why not moving this to `macrosDepsByVersion`? Unless I am blinder than what I believe both are always sued together so I would rather just have one.
WDYT?

Reply:
Sounds good, I will aggregate these "byVersion" functions.